Raiser/IncludedFileInfo.cpp

```diff
@@ -151,16 +151,16 @@ class FuncDeclFinder : public clang::ASTConsumer {
 
   void HandleTranslationUnit(clang::ASTContext &Context) final {
     auto Decls = Context.getTranslationUnitDecl()->decls();
-    clang::SourceManager &SourceManager(Context.getSourceManager());
+    // clang::SourceManager &SourceManager(Context.getSourceManager());
     for (auto &Decl : Decls) {
       if (Decl->isFunctionOrFunctionTemplate()) {
-        const auto &FileID = SourceManager.getFileID(Decl->getLocation());
-        if (FileID != SourceManager.getMainFileID())
-          continue;
+        // const auto &FileID = SourceManager.getFileID(Decl->getLocation());
+        // if (FileID != SourceManager.getMainFileID())
+        //   continue;
```

Comment:
Why was this commented out?

Reply:
With this lines scan function prototipes **only** in this file, internal #include visitor ignored.
But if defined sysroot and target and exists toolchain with headers tooling loads all internal includs.

Without this line may define header file like this

```bash
// header-inc.h

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
```

Reply:
The code commented out gets the `FileID` of the candidate `Decl` - which `isFunctionOrFunctiontemplate`. The `if` condition checks to ensure `Decl` is visited (and considered for addition to the list of external/known function prototypes) only if it belongs to the main file (such as `/usr/inculde/stdio.h`).

I don't see how this code is involved in skipping over internal `#include`s. Can you please add some code here using `LLVM_DEBUG` (such as on [line 81](https://github.com/sv99/llvm-mctoll/blob/d45c4fae5147f12284191f8e18efabf1515c6f32/Raiser/IncludedFileInfo.cpp#L81) of this file) to demonstrate your observation. Thx!

Reply:
Problem with `if (FileID != SourceManager.getMainFileID())`
if using my  header-inc.h file (instead of `/usr/inculde/stdio.h`) list function is empty!!

With commented lines - include all files from  <stdio.h>,  <stdlib.h> and <string.h>

Reply:
OK. I believe [this patch](https://github.com/microsoft/llvm-mctoll/files/9539276/incf.txt) addresses the issue appropriately. I've checked its correctness by 

a) building the target `check-mctoll` and 
b) specifying a `header-inc.h` that includes other include files, as you indicated.

Please apply it and let me know if you have any questions. Thx!